In [1]:
import pandas as pd

In [5]:
dataset=pd.read_json('data/All_Beauty_5.json', lines=True)
dataset.head(10)

overall  verified   reviewTime      reviewerID        asin  \
0        5      True   09 1, 2016  A3CIUOJXQ5VDQ2  B0000530HU   
1        5      True  11 14, 2013  A3H7T87S984REU  B0000530HU   
2        1      True  08 18, 2013  A3J034YH7UG4KT  B0000530HU   
3        5     False   05 3, 2011  A2UEO5XR3598GI  B0000530HU   
4        5      True   05 6, 2011  A3SFRT223XXWF7  B00006L9LC   
5        5     False  05 16, 2010  A24HQ2N7332W7W  B00006L9LC   
6        5     False   05 7, 2018  A2G90R2ZU6KU5D  B00006L9LC   
7        2      True   05 7, 2018  A24W4W9E62FZP2  B00006L9LC   
8        1      True   05 6, 2018   A7ID5H7FWLJHC  B00006L9LC   
9        1      True   05 6, 2018   AYKOSAJTP5AVS  B00006L9LC   

                                               style  \
0  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...   
1  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...   
2  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...   
3  {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...   
4                          {'Size:': ' 200ml/6.7oz'}   
5                                  {'Size:': ' 366'}   
6                                {'Size:': ' Small'}   
7                                {'Size:': ' Small'}   
8                                {'Size:': ' Small'}   
9                                {'Size:': ' Small'}   

                   reviewerName  \
0                      Shelly F   
1                  houserules18   
2                          Adam   
3                        Rich K   
4               C. C. Christian   
5  Kindle Customer Joyce Wilson   
6                          Mike   
7                           Reb   
8                         U. V.   
9               Senthil Kumar M   

                                          reviewText  \
0                   As advertised. Reasonably priced   
1  Like the oder and the feel when I put it on my...   
2  I bought this to smell nice after I shave.  Wh...   
3  HEY!! I am an Aqua Velva Man and absolutely lo...   
4  If you ever want to feel pampered by a shampoo...   
5  If you know the scent of Diva, you'll LOVE thi...   
6  Got this shampoo as a solution for my wife's d...   
7  No change my scalp still itches like crazy. It...   
8  Too expensive for such poor quality. There was...   
9  It dries my hair, doesnt help to reduce dandru...   

                                             summary  unixReviewTime vote  \
0                                         Five Stars      1472688000  NaN   
1                                  Good for the face      1384387200  NaN   
2                                       Smells awful      1376784000  NaN   
3  Truth is There IS Nothing Like an AQUA VELVA MAN.      1304380800   25   
4                                    Bvlgari Shampoo      1304640000    3   
5                                   Diva is Heavenly      1273968000  NaN   
6                          Outstanding, no complains      1525651200  NaN   
7  No change my scalp still itches like crazy. It...      1525651200  NaN   
8  Too expensive for such poor quality. There was...      1525564800  NaN   
9  Dries my hair, doesnt help to reduce dandruff....      1525564800  NaN   

  image  
0   NaN  
1   NaN  
2   NaN  
3   NaN  
4   NaN  
5   NaN  
6   NaN  
7   NaN  
8   NaN  
9   NaN

In [7]:
dataset['style']

0       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
1       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
2       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
3       {'Size:': ' 7.0 oz', 'Flavor:': ' Classic Ice ...
4                               {'Size:': ' 200ml/6.7oz'}
                              ...                        
5264                       {'Design:': ' ETA-C6B5F7C374'}
5265                       {'Design:': ' ETA-7117EE3788'}
5266                       {'Design:': ' ETA-BF66BD2F87'}
5267                       {'Design:': ' ETA-03868FCB34'}
5268                       {'Design:': ' ETA-C4893097E9'}
Name: style, Length: 5269, dtype: object

In [18]:
dataset.drop(['style','image'], axis=1).describe(include='all') #style and image are causing issues with describe()

overall verified   reviewTime     reviewerID        asin  \
count   5269.000000     5269         5269           5269        5269   
unique          NaN        2          883            991          85   
top             NaN     True  07 14, 2016  AKJHHD5VEH7VG  B0012Y0ZG2   
freq            NaN     4280           33             12        1794   
mean       4.771873      NaN          NaN            NaN         NaN   
std        0.743204      NaN          NaN            NaN         NaN   
min        1.000000      NaN          NaN            NaN         NaN   
25%        5.000000      NaN          NaN            NaN         NaN   
50%        5.000000      NaN          NaN            NaN         NaN   
75%        5.000000      NaN          NaN            NaN         NaN   
max        5.000000      NaN          NaN            NaN         NaN   

           reviewerName     reviewText     summary  unixReviewTime vote  
count              5269           5264        5264    5.269000e+03  552  
unique              905           1363        1011             NaN   24  
top     Amazon Customer  Great product  Five Stars             NaN    2  
freq                362             15        1455             NaN  238  
mean                NaN            NaN         NaN    1.441506e+09  NaN  
std                 NaN            NaN         NaN    5.473127e+07  NaN  
min                 NaN            NaN         NaN    1.095725e+09  NaN  
25%                 NaN            NaN         NaN    1.407024e+09  NaN  
50%                 NaN            NaN         NaN    1.445645e+09  NaN  
75%                 NaN            NaN         NaN    1.481674e+09  NaN  
max                 NaN            NaN         NaN    1.534982e+09  NaN

The dataset contains 5264 reviews for 85 Beauty products from 991 reviewers